class-es-wp-query-shoehorn.php

```diff
@@ -218,5 +224,15 @@ private function reboot_query_vars( &$query ) {
 		if ( empty( $q['posts_per_page'] ) ) {
 			$q['posts_per_page'] = get_option( 'posts_per_page' );
 		}
+
+		// Restore the author ID which is normally added during get_posts() in WP_Query.
+		// Required for handle_404() in WP class to not mark empty author archives as 404s.
+		if ( $query->is_author() && ! empty( $q['author_name'] ) ) {
```

Comment:
This restores the core functionality from [`get_posts()` in `WP_Query`](https://github.com/WordPress/WordPress/blob/4b0e64aa8c8723781a2428160d3d9c18a0e0d3cd/wp-includes/class-wp-query.php#L2137-L2139).

Reply:
For consistency with the `get_posts()` method, we should include the handling for author names with slashes in them, which appears just above the lines you've referenced.